pubsub/topic.go

```diff
@@ -51,6 +51,14 @@ const (
 // ErrOversizedMessage indicates that a message's size exceeds MaxPublishRequestBytes.
 var ErrOversizedMessage = bundler.ErrOversizedItem
 
+// QueueError indicates custom error for testing purposes
+var QueueError error
```

Comment:
Why not to use it as a server object field?

Reply:
`r := topic.Publish(ctx, &Message{})`
The Publish function uses Context, Message, and Topic. Context and Message don't contain the server, while the topic has Client, Client has Publisher and Subscriber, no server here too. I can't find a straightforward way to use server object fields for this function

Reply:
But wait a bit, you're adding test feature into real functionality. This isn't good, and not something Google will accept. The feature should be implemented as close to pstest as it can be. For now you're adding public function into the real PubSub file, while it should be implemented only for pstest, only for test cases. it should be done in pstest file at least. It is a pstest feature request.